apps/judicial-system/backend/src/app/modules/event-log/eventLog.service.ts

```diff
@@ -72,28 +74,24 @@ export class EventLogService {
     event: CreateEventLogDto,
     transaction?: Transaction,
   ): Promise<boolean> {
-    const {
-      eventType,
-      caseId,
-      userName,
-      userRole,
-      nationalId,
-      institutionName,
-    } = event
+    const { eventType, caseId, userName, userRole, institutionName } = event
 
     if (!allowMultiple.includes(event.eventType)) {
       const where = Object.fromEntries(
         // The user name and title do not matter when checking for duplicates
         Object.entries({
           eventType,
           caseId,
-          nationalId,
-          userRole,
-          institutionName,
+          userRole: allowOnePerUserRole.includes(eventType)
+            ? userRole
+            : undefined,
         }).filter(([_, value]) => value !== undefined),
       )
 
-      const eventExists = await this.eventLogModel.findOne({ where })
+      const eventExists = await this.eventLogModel.findOne({
+        where,
+        transaction,
```

Comment:
We want to use the same transaction here as in the event log creation to prevent accidental creation of duplicate events.